### PR TITLE
Imagecollector crd

### LINKED ISSUE #165: Create ImageCollector CRD
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

Add ImageCollector CRD to hold all collected images as well as the scanner results.
- For CR for each node -  Spec: all images on node
- For final shared CR - Spec: deduplicated list of all images, Status: scanner output of vulnerable images

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Eraser version:
- Kubernetes version: (use `kubectl version`):

### PR DESCRIPTION
**Pull request by ashnamehrotra**
**What this PR does / why we need it**:
Defines ImageCollector CR.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #165 

**Special notes for your reviewer**:

### CODE REVIEW

File: api/v1alpha1/imagecollector_types.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+)
+
+type ImageCollectorSpec struct {
+	Images []Image `json:"images"`
+}
+
+type Image struct {
+	Digest string   `json:"digest"`
+	Name   []string `json:"name,omitempty"`
+	Node   string   `json:"node,omitempty"`
```
Comment (pmengelbert) [inline api/v1alpha1/imagecollector_types.go]: We can remove `Node` here

File: api/v1alpha1/imagecollector_types.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ImageCollectorSpec struct {
+	Images []Image `json:"images"`
+}
+
+type Image struct {
+	Digest string   `json:"digest"`
+	Name   []string `json:"name,omitempty"`
```
Comment (pmengelbert) [inline api/v1alpha1/imagecollector_types.go]: Just a simple string here, and the collector pod selects the first name.

**Comment by pmengelbert** [comment]
The CRD looks right, but `make manifests` doesn't auto-generate the CRD. I think there is some piece of kubebuilder config missing.
```bash
$ make manifests deploy
<...>
$ kubectl get customresourcedefinitions.apiextensions.k8s.io
NAME                   CREATED AT
imagejobs.eraser.sh    2022-05-09T21:40:07Z
imagelists.eraser.sh   2022-05-09T21:40:07Z
```

**Comment by sozercan** [comment]
You need to add to https://github.com/Azure/eraser/blob/44bde01c1e408d2aeed161d9b568de6db79ac8af/config/crd/kustomization.yaml#L6

**Review by pmengelbert — APPROVED** [review]
LGTM

**Review by sozercan — APPROVED** [review]

**Pull request merged**